src/gar.lua

```diff
@@ -185,15 +209,26 @@ function gar.decreaseOperatorStake(from, qty, currentTimestamp, msgId)
 				.. " IO"
 		)
 	end
+	gateway.operatorStake = gar.getGateway(from).operatorStake - qty
+
+	if instantWithdraw == true then
+		-- Calculate the penalty and withdraw using the utility function
+		-- TO DO: RETURN THESE!
+		expeditedWithdrawalFee, amountToWithdraw = processInstantWithdrawal(nil, 0, 0, from, qty)
+	else
+		gateway.vaults[msgId] = {
+			balance = qty,
+			startTimestamp = currentTimestamp,
+			endTimestamp = currentTimestamp + gar.getSettings().operators.withdrawLengthMs,
+		}
+	end
 
+	-- Update the operator stake if not an instant withdrawal
 	gateway.operatorStake = gar.getGateway(from).operatorStake - qty
-	gateway.vaults[msgId] = {
-		balance = qty,
-		startTimestamp = currentTimestamp,
-		endTimestamp = currentTimestamp + gar.getSettings().operators.withdrawLengthMs,
-	}
-	-- update the gateway
+
+	-- Update the gateway
 	GatewayRegistry[from] = gateway
+
 	return gar.getGateway(from)
```

Comment:
Should be pretty straightforward to update this to return a table containing the gateway and the penalty info that we want to event on later.

Reply:
Updated decreaseOperatorStake and decreaseDelegateStake

https://github.com/ar-io/ar-io-network-process/commit/1e636ccb7984857e9dde09385f64171752a39d75